streams/src/test/java/org/apache/kafka/streams/integration/SuppressionIntegrationTest.java

```diff
@@ -0,0 +1,319 @@
+            for (final KVT<String, String> record : toProduce) {
+                final Future<RecordMetadata> f = producer.send(
+                    new ProducerRecord<>(topic, null, record.timestamp, record.key, record.value, null)
+                );
+                futures.add(f);
+            }
+            for (final Future<RecordMetadata> future : futures) {
+                try {
+                    future.get();
+                } catch (final InterruptedException | ExecutionException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+            // TODO: test EOS
+            //noinspection ConstantConditions
+            if (false) {
+                producer.commitTransaction();
+            }
+            producer.flush();
+        }
+    }
+
+    private void verifyOutput(final String topic, final List<KVT<String, Long>> expected) {
+        final List<ConsumerRecord<String, Long>> results = drainConsumerRecords(topic, STRING_DESERIALIZER, LONG_DESERIALIZER, expected.size());
+        if (results.size() != expected.size()) {
+            throw new AssertionError(printRecords(results) + " != " + expected);
+        }
+        final Iterator<KVT<String, Long>> expectedIterator = expected.iterator();
+        for (final ConsumerRecord<String, Long> result : results) {
+            final KVT<String, Long> expected1 = expectedIterator.next();
+            try {
+                compareKeyValueTimestamp(result, expected1.key, expected1.value, expected1.timestamp);
+            } catch (final AssertionError e) {
+                throw new AssertionError(printRecords(results) + " != " + expected, e);
+            }
+        }
+    }
+
+    private <K, V> List<ConsumerRecord<K, V>> drainConsumerRecords(final String topic, final Deserializer<K> keyDeserializer, final Deserializer<V> valueDeserializer, final int numRecords) {
+        final KafkaConsumer<K, V> consumer = new KafkaConsumer<>(
+            mkProperties(
+                mkMap(
+                    mkEntry(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers()),
+                    mkEntry(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, keyDeserializer.getClass().getName()),
+                    mkEntry(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, valueDeserializer.getClass().getName())
+                )
+            )
+        );
+        final TopicPartition topicPartition = new TopicPartition(topic, 0);
+        consumer.assign(singletonList(topicPartition));
+        consumer.seekToBeginning(singletonList(topicPartition));
+        final List<ConsumerRecord<K, V>> results = new LinkedList<>();
+        while (results.size() < numRecords) {
+            final ConsumerRecords<K, V> consumerRecords = consumer.poll(Duration.ofMillis(100));
+            for (final ConsumerRecord<K, V> consumerRecord : consumerRecords) {
+                System.out.println(consumerRecord);
```

Comment:
was this line intentional or leftover debugging?

Reply:
Doh! This is left over. Sorry.